### ISSUE TITLE
[FR] Trigorilla Pro STM32 - Power loss detection

### ISSUE THREAD

**Issue by van0014** [MarlinFirmware/Marlin#27244]
### Is your feature request related to a problem? Please describe.

There is a bit of information on the Trigorilla Pro board, and it seems practical to implement this. There's a nice pin debug system in Marlin, and this was enough for me to do some tests with. I just couldn't implement it.

I would try to do it myself, but am not experienced enough with C. Only with Arduino, which isn't as advanced.

### Are you looking for hardware support?

Anycubic Delta - Trigorilla Pro, STM32

### Describe the feature you want

Power loss detection with the stock Trigorilla Pro, STM32 hardware

The hardware may not yet be fully understood. The power loss circuit must first be activated before the onboard capacitor is charged. And i'm not certain whether the capacitor or FET needs to be soft started

### Additional context

There is some schematics on thingyverse, as well as pin function explainations.

https://www.thingiverse.com/thing:4271636

    -The cap charge/activate pin is on Pin PG4 (Pin 100)
    -The comparator is on Pin PG2 (Pin 98). Pin seems to be LOW normally, and HIGH when 24v power is off

**Comment by rondlh** [comment]
I don't fully understand, what would be the benefit of this feature?
Don't these backup power modules have a power fail signal/pin, that is used by Marlin to handle some last minute activities?

**Comment by van0014** [comment]
Yes, so basically power loss isn't working with stock trigorilla pro hardware. They have a pin to signal power loss. They also have a 1F capacitor, but it's FET driven. 

If it's safe, the FET can just be turned on when power loss detection is needed. Otherwise, a basic PWM soft start should be all it needs. It seems that once it's charged, it should continue powering the 5v rail via a FET that's powered from the charged capacitor. Possibly the power loss circuit just needs a quick pulse, before there's enough voltage in the cap for the 5v side FET to remain on

The comparator may only work properly once the 'charge' pin is off again, because the cap 'charge' pin is linked to the comparator positive.

I've done some basic tests with Marlin's debug feature. Unfortunately because I changed my stepper motors, I cannot easily run stock firmware to properly test how it functions when charging and during power loss

![image](https://github.com/user-attachments/assets/157619c3-bfa4-4727-826c-9d303f7ae2d2)
